packages/grid/_modules_/grid/utils/utils.ts

```diff
@@ -22,7 +22,8 @@ export function getThemePaletteMode(palette: any): string {
 }
 
 export function isMuiV5(): boolean {
-  return 'alpha' in styles;
+  // Changed from `alpha` to 'fade' because mui/core@4.12.0 adds 'alpha' in v4
```

Comment:
Could we check several properties to reduce the risk of a future break ?

Reply:
Nice trick but this looks like a comment for a the PR's description, not the source

```suggestion
```